### PR TITLE
angular2 cutoff

### LINKED ISSUE #11: README.md suggests quartermaster is fully operational after setting up IntelliJ support
Running `$ ./gradlew -t classes` after cloning repo (as README.md suggests) leads to errors since infiniboard-app has no valid npm configuration.

(Do not know why, :infiniboard-app:node-setup seems to be inserted too late in the dependency hierarchy and is never called.)

`$ (cd infiniboard-app && npm install)` beforehand fixes the issue.

Also, please add that a Python in [2.5, 3.0) must be available on the system and that gulp-cli should have been installed globally (on npm).

### LINKED ISSUE #18: fix angular2 startup after using gulp
When using `npm start` the app works as expected and angular2 loads. But after using `gulp` to concat / minify css and js angular2 doesn't load.

### LINKED ISSUE #26: configure quartermaster tomcat for angular2
In order for angular2 HTML5 navigation to work properly, all angular2 app request must be redirected to index.html.

An example on how to configure this redirect for apache2 can be found here:
http://serverfault.com/questions/188373/apache-2-2-rewrite-force-all-requests-to-index-html

Also the quartermaster `/api/**/*` requests have to be routed to spring boot app!

### PR DESCRIPTION
**Pull request by matthiasbalke**
This PR contains the cutoff through the angular app.
 
Fixes
- [x] fixed gulp typescript compiling (closes #18)
- [x] fixed `README.md` (closes #11)
- [x] added angular2 Router to navigate between Dashboards
- [x] configure tomcat for angular2 routing (closes #26)

Structure
- [x] added a Sidebar navigation for Dashboards

Features
- [x] added a Service for Dashboards
- [x] added platform-status widget
- [x] added jenkins job status widget
- [x] load all dashboards
- [x] load all widgets for selected dashboard
- [x] load widget config and init widget
- [x] update widget
- [x] poll widget data updates

It will contain one widget which pulls the data from a service which in turn uses mock data instead of calling the `quartermaster` backend.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,8 +36,8 @@ bower_components
 typings
 
 # transpiled typescript
-infiniboard-app/app/*.js
-infiniboard-app/app/*.js.map
+infiniboard-app/**/*.js
```
Comment (RudiKlassen) [inline .gitignore]: Should not be necessary, as typescript gets compiled to `build/app` now.
This expression also matches `node_modules/**/*.js` which is not intended.

Just revert change.

Comment (matthiasbalke) [inline .gitignore]: ok I removed it.

**Comment by matthiasbalke** [comment]
This version is ready for review and merge. Unittesting and others will be added in later PRs.

**Comment by RudiKlassen** [comment]
Merged the PR. Cheers mate.

**Pull request merged**